lisp/ess-mode.el

```diff
@@ -429,32 +421,39 @@ current function."
 
 
 ;;; Formatting / indentation
+(defvar-local ess--installed-style-vars nil
+  "A cons of the form (STYLE . VARS).
+VARS is a list of all style vars which were not set explicitly to
+buffer local values by the user in mode hooks.")
 
-(defun ess-set-style (&optional style quiet)
+(defun ess-set-style (&optional style _quiet)
   "Set up the `ess-mode' style variables from the `ess-style' variable.
 If STYLE argument is given, use that instead. It makes the ESS
-indentation style variables buffer local. When non-nil, QUIET
-suppresses messaging."
-  (interactive)
-  (let ((ess-styles (mapcar 'symbol-name (mapcar 'car ess-style-alist))))
-    (unless style
-      (setq style
-            (intern (ess-completing-read "Set ESS mode indentation style"
-                                         ess-styles nil t nil nil ess-style))))
-    (setq style (or style ess-style))
-    (make-local-variable 'ess-style)
-    (if (memq (symbol-name style) ess-styles)
-        (setq ess-style style)
-      (error (format "Bad ESS style: %s" style)))
-    (if (not quiet)
-        (message "ESS-style: %s" ess-style))
-    ;; finally, set the indentation style variables making each one local
-    (mapc (lambda (ess-style-pair)
-            (make-local-variable (car ess-style-pair))
-            (set (car ess-style-pair)
-                 (cdr ess-style-pair)))
-          (cdr (assq ess-style ess-style-alist)))
-    ess-style))
+indentation style variables buffer local. QUIET is for backward
+compatibility and is ignored."
+  (interactive
+   (list (let ((styles (mapcar (lambda (x) (symbol-name (car x)))
+                               ess-style-alist)))
+           (intern (ess-completing-read
+                    "Set ESS mode indentation style"
+                    styles nil t nil nil ess-style)))))
+  (let* ((keep-local (null style))
+         (style (or style ess-style))
+         (style-alist (or (cdr (assq style ess-style-alist))
+                          (error (format "Bad ESS style: %s" style))))
```

Comment:
The string argument of error gets passed to `format-message`, so no need to call `format` here, I think.